src/main/java/com/redhat/devtools/lsp4ij/features/documentation/LSPDocumentationHelper.java

```diff
@@ -26,63 +30,115 @@ private LSPDocumentationHelper() {
-        if (htmlBody == null || htmlBody.isEmpty()) {
-            return htmlBody;
+    public static @NotNull List<MarkupContent> getValidMarkupContents(@Nullable Hover hover) {
+        if (hover == null) {
+            // textDocument/hover may return null
+            return Collections.emptyList();
         }
-        Color background = editor != null ? editor.getColorsScheme().getDefaultBackground() : null;
-        Color foreground = editor != null ? editor.getColorsScheme().getDefaultForeground() : null;
-
-        StringBuilder html = new StringBuilder("<html><head><style TYPE='text/css'>html { ");
-        if (background != null) {
-            html.append("background-color: ")
-                    .append(toHTML(background))
-                    .append(";");
+        var hoverContents = hover.getContents();
+        if (hoverContents == null) {
+            return Collections.emptyList();
         }
-        if (foreground != null) {
-            html.append("color: ")
-                    .append(toHTML(foreground))
-                    .append(";");
+        if (hoverContents.isLeft()) {
+            var contents = hoverContents.getLeft();
+            if (contents == null || contents.isEmpty()) {
+                return Collections.emptyList();
+            }
+            return contents.stream()
+                    .map(content -> {
+                        if (content.isLeft()) {
+                            String value = content.getLeft();
+                            if (!isValidContent(value)) {
+                                return null;
+                            }
+                            return new MarkupContent(MarkupKind.MARKDOWN, value);
+                        } else if (content.isRight()) {
+                            MarkedString markedString = content.getRight();
+                            if (!isValidContent(markedString)) {
+                                return null;
+                            }
+                            String value = markedString.getValue();
+                            String language = markedString.getLanguage() != null && !markedString.getLanguage().isEmpty() ?
+                                    markedString.getLanguage() :
+                                    null;
+                            if (language == null) {
+                                return new MarkupContent(MarkupKind.MARKDOWN, value);
+                            }
+                            value = String.format("```%s%n%s%n```", language, value);
+                            return new MarkupContent(MarkupKind.MARKDOWN, value);
+                        }
+                        return null;
+                    })
+                    .filter(Objects::nonNull)
+                    .toList();
+        } else if (hoverContents.isRight()){
+            MarkupContent content = hoverContents.getRight();
+            if (!isValidContent(content)) {
+                return null;

```

Comment:
causes an exception, when clicking on "reset go.mod diagnostic" codelens in a go.mod file:

```
java.lang.IllegalStateException: @NotNull method com/redhat/devtools/lsp4ij/features/documentation/LSPDocumentationHelper.getValidMarkupContents must not return null
	at com.redhat.devtools.lsp4ij.features.documentation.LSPDocumentationHelper.$$$reportNull$$$0(LSPDocumentationHelper.java)
	at com.redhat.devtools.lsp4ij.features.documentation.LSPDocumentationHelper.getValidMarkupContents(LSPDocumentationHelper.java:102)
	at com.redhat.devtools.lsp4ij.features.documentation.LSPDocumentationTargetProvider.toDocumentTarget(LSPDocumentationTargetProvider.java:105)
	at com.redhat.devtools.lsp4ij.features.documentation.LSPDocumentationTargetProvider.lambda$documentationTargets$0(LSPDocumentationTargetProvider.java:89)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at com.redhat.devtools.lsp4ij.features.documentation.LSPDocumentationTargetProvider.documentationTargets(LSPDocumentationTargetProvider.java:91)
```

Reply:
fixed with tests